src/transformState.js

```diff
@@ -5,7 +5,37 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  let records = {};
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const data in state) {
+          delete state[data];
+        }
+        break;
+      case 'addProperties':
+        records = action.extraData;
+
+        for (const record in records) {
+          state[record] = records[record];
+        }
+        break;
+      case 'removeProperties':
+        const removes = action.keysToRemove;
+
+        for (const remove of removes) {
+          for (const data in state) {
+            if (remove === data) {
+              delete state[data];
+            }
+          }
+        }
```

Comment:
```suggestion
        for (const key of action.keysToRemove) {
          delete state[key];
        }
```
Тут також не потрібно два цикли. В тебе є масив з ключами, які потрібно видалити. От по ньому і варто лише пройтись, видаливши у об'єкті state всі необхідні ключі